### ISSUE TITLE
Migrate deployment from Travis to Actions

### ISSUE THREAD

**Issue by kcranston** [earthlab/hub-ops#313]
Steps for migration:

- [ ] workflow for building and pushing docker images, see [template for docker build + push](https://github.com/actions/starter-workflows/blob/edab90608156748407e4aa995dca600203ab03f2/ci/docker-image.yml)
- [ ] workflow for deploying new clusters (helm updates), see [template for GKE deployments](https://github.com/actions/starter-workflows/blob/edab90608156748407e4aa995dca600203ab03f2/ci/google.yml)
- [ ] change the way we do deployment secrets by adding them to the repo under `Settings` rather than storing them as code in the repo; see https://docs.github.com/en/free-pro-team@latest/actions/reference/encrypted-secrets  

@mbjoseph has given us a good first step by translating our travis file to Actions [in his fork](https://github.com/mbjoseph/hub-ops/actions). We probably want to aim for re-writing the whole workflow in Actions (i.e. use as much of the Actions tooling as possible, rather than just calling `deploy.py` script in an Actions workflow and having it do most of the work).

**Comment by lwasser** [comment]
@kcranston i've been adding secrets to the repos to support actions. let me know what you need as it's super  easy to do. i'm really loving actions  a lot

**Comment by lwasser** [comment]
the one consideration is that if deploy.py works, it could be run on any CI infrastructure so the question is - are we  sure we want to stick with actions for a while or is flexibilty nice. i also run deploy locally and it seems to work  when i'm in a pinch - ci not working.

**Comment by kcranston** [comment]
Good point, @lwasser We will need to make changes to `deploy.py` to remove the travis bits, so this might be a good time to add a --local option for running locally using local gcloud credentials.

**Comment by lwasser** [comment]
@kcranston i think we can close  this. can you close it if we  can?
